common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -1434,6 +1435,42 @@ public static UTF8String[] icuSplitSQL(final UTF8String string, final UTF8String
     return strings.toArray(new UTF8String[0]);
   }
 
+  /**
+   * Splits the `string` into an array of substrings based on the `delimiter` regex, with respect
+   * to the maximum number of substrings `limit`.
+   *
+   * @param string the string to be split
+   * @param delimiter the delimiter regex to split the string
+   * @param limit the maximum number of substrings to return
+   * @return an array of substrings
```

Comment:
Please, start the comments from upper cased letters.

Reply:
This has not been the practice for this file. Most of the documentation comments here start with a lowercase. It might be more cleaner to do a followup PR to adjust all those accordingly.